model/app.py

```diff
@@ -222,24 +262,28 @@ def key_from_model_info(wild_type_id, message):
     """
     d = {k: message.get(k, []) for k in {GENOTYPE_CHANGES, MEDIUM, MEASUREMENTS, REACTIONS_ADD, REACTIONS_KNOCKOUT}}
     d['model_id'] = wild_type_id
+
+    if version:
+        d['version'] = version
+
     return hashlib.sha224(json.dumps(d, sort_keys=True).encode('utf-8')).hexdigest()
 
 
-async def save_changes_to_db(model, wild_type_id, message):
+async def save_changes_to_db(model, wild_type_id, message, version=None):
     """Store model in cache database
 
     :param model: Cameo model
     :param wild_type_id: str
     :param message: dict
-    :return: str (cache database key)
+    :return: mutated_model_id (cache database key)
     """
-    db_key = key_from_model_info(wild_type_id, message)
+    mutated_model_id = key_from_model_info(wild_type_id, message, version)
     redis = await redis_client()
```

Comment:
At least the redis connection pool also allows for a context. See here http://aioredis.readthedocs.io/en/v0.3.0/start.html#python-3-5-async-with-async-for-support

Reply:
I don't quite get your point. I bumped the new redis version locally, seems to work fine. Do we need blocking?

Reply:
Sorry, my point was that instead of the current code which is basically `open()`, `modify()`, and `close()`, we could use something like:
```python
with (await pool) as conn:
    value = await conn.execute('get', 'my-key')
```
which does the same thing really but makes it impossible to forget to close a connection and looks nicer IMO since it mirrors the file handling syntax.